[2025-07-22T05:32:55.248+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-07-22T05:32:55.276+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: weather_dag.fetch_forecast manual__2025-07-22T05:32:52.416376+00:00 [queued]>
[2025-07-22T05:32:55.283+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: weather_dag.fetch_forecast manual__2025-07-22T05:32:52.416376+00:00 [queued]>
[2025-07-22T05:32:55.284+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 3
[2025-07-22T05:32:55.295+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): fetch_forecast> on 2025-07-22 05:32:52.416376+00:00
[2025-07-22T05:32:55.303+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'weather_dag', 'fetch_forecast', 'manual__2025-07-22T05:32:52.416376+00:00', '--job-id', '107', '--raw', '--subdir', 'DAGS_FOLDER/weather_dag.py', '--cfg-path', '/tmp/tmp83ldaubp']
[2025-07-22T05:32:55.306+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=178) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-07-22T05:32:55.307+0000] {standard_task_runner.py:91} INFO - Job 107: Subtask fetch_forecast
[2025-07-22T05:32:55.307+0000] {standard_task_runner.py:63} INFO - Started process 180 to run task
[2025-07-22T05:32:55.356+0000] {task_command.py:426} INFO - Running <TaskInstance: weather_dag.fetch_forecast manual__2025-07-22T05:32:52.416376+00:00 [running]> on host 791cfe7304d9
[2025-07-22T05:32:55.425+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='weather_dag' AIRFLOW_CTX_TASK_ID='fetch_forecast' AIRFLOW_CTX_EXECUTION_DATE='2025-07-22T05:32:52.416376+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-07-22T05:32:52.416376+00:00'
[2025-07-22T05:32:55.427+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-07-22T05:32:55.436+0000] {weather_dag.py:53} INFO - Bắt đầu gọi API thời tiết...
[2025-07-22T05:33:05.966+0000] {weather_dag.py:98} INFO - Các cột có giá trị None: name                           0
forecast_time                  0
temperature                    0
temperatureApparent            0
dewPoint                       0
humidity                       0
windSpeed                      0
windDirection                  0
windGust                       0
pressureSurfaceLevel           0
cloudCover                     0
precipitationProbability       0
uvIndex                      160
ezHeatStressIndex           1200
dtype: int64
[2025-07-22T05:33:05.973+0000] {weather_dag.py:102} WARNING - Cột name vẫn chứa chuỗi 'nan'!
[2025-07-22T05:33:05.993+0000] {weather_dag.py:102} WARNING - Cột uvIndex vẫn chứa chuỗi 'nan'!
[2025-07-22T05:33:06.024+0000] {weather_dag.py:114} INFO - name                        object
forecast_time               object
temperature                 object
temperatureApparent         object
dewPoint                    object
humidity                     int64
windSpeed                   object
windDirection                int64
windGust                    object
pressureSurfaceLevel        object
cloudCover                   int64
precipitationProbability     int64
uvIndex                     object
ezHeatStressIndex           object
dtype: object
[2025-07-22T05:33:06.772+0000] {weather_dag.py:137} INFO - Đã insert 1200 dòng dữ liệu thời tiết
[2025-07-22T05:33:06.775+0000] {python.py:237} INFO - Done. Returned value was: None
[2025-07-22T05:33:06.776+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-07-22T05:33:06.786+0000] {taskinstance.py:1206} INFO - Marking task as SUCCESS. dag_id=weather_dag, task_id=fetch_forecast, run_id=manual__2025-07-22T05:32:52.416376+00:00, execution_date=20250722T053252, start_date=20250722T053255, end_date=20250722T053306
[2025-07-22T05:33:06.805+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 0
[2025-07-22T05:33:06.821+0000] {taskinstance.py:3503} INFO - 1 downstream tasks scheduled from follow-on schedule check
[2025-07-22T05:33:06.823+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
